src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -792,7 +792,8 @@ G1PreConcurrentStartTask::G1PreConcurrentStartTask(GCCause::Cause cause, G1Concu
 void G1ConcurrentMark::pre_concurrent_start(GCCause::Cause cause) {
   assert_at_safepoint_on_vm_thread();
 
-  G1CollectedHeap::start_codecache_marking_cycle_if_inactive();
+  CodeCache::on_gc_marking_cycle_start();
+  CodeCache::arm_all_nmethods();
```

Comment:
Maybe factor this out into a method (`start_codecache_marking_cycle`) and call it above.

Reply:
Like this right?

```c++
void G1CollectedHeap::start_codecache_marking_cycle() {
  CodeCache::on_gc_marking_cycle_start();
  CodeCache::arm_all_nmethods();
}
```

Reply:
Exactly. Also I meant "... to call that method here and above"; i.e. I asked to factor out because this is done twice.

Reply:
Ok, will do.

Reply:
Maybe the "finish" stuff could be extracted out similarly, I did not look if the calls are in the same file so it may not be possible.